includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2155,13 +2155,18 @@ public function add_twentytwentyone_dark_mode_toggle() {
 	 * Make the mobile menu for the Twenty Twenty-One theme AMP compatible.
 	 */
 	public function add_twentytwentyone_mobile_modal() {
-		$menu_query  = $this->dom->xpath->query( "//div[ @class and contains( concat( ' ', normalize-space( @class ), ' ' ), ' primary-menu-container ' ) ]" );
-		$menu_toggle = $this->dom->getElementById( 'primary-mobile-menu' );
+		$menu_query        = $this->dom->xpath->query( "//div[ @class and contains( concat( ' ', normalize-space( @class ), ' ' ), ' primary-menu-container ' ) ]" );
+		$menu_toggle_query = $this->dom->xpath->query( "//button[ @id = 'primary-mobile-menu' and ./span[ contains( concat( ' ', normalize-space( @class ), ' ' ), ' open ' ) ] and ./span[ contains( concat( ' ', normalize-space( @class ), ' ' ), ' close ' ) ] ]" );
```

Comment:
The predicate here seems to be a bit too specific. I think it could be:

```suggestion
		$menu_toggle_query = $this->dom->xpath->query( "//button[ @id = 'primary-mobile-menu' ]" );
```

And then below we can `return` if `$menu_open_toggle` or `$menu_close_toggle` are null.